### PR TITLE
merge release_23.0 into dev

### PR DESCRIPTION
**Pull request by martenson**
resolved some conflict in code from @mvdbeek @jmchilton -- please sanity check
## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Review by jmchilton — APPROVED** [review]
Sanity checked unless CI reveals I am wrong.

**Comment by martenson** [comment]
There is an issue in compiling .ts that I'd expect to work with duck typing but it does not seem to be the case.

**Comment by ElectronicBlueberry** [comment]
`step.id` for `const step = ...` is of type `number | undefined`, since `getStep` potentially returns undefined.

`assertDefined` ensures `step` is defined, however since a spread is used here, `step` is always defined, but its values remain potentially undefined. The spread must be preformed after the assertion, so that this potential error can be caught correctly.

```ts
    let step = stepStore.getStep(stepId.value);
    assertDefined(step);
    step = { ...step };
```

or, a bit safer

```ts
    const sourceStep = stepStore.getStep(stepId.value);
    assertDefined(sourceStep);
    const step = { ...sourceStep };
```

File: client/src/components/Workflow/Editor/NodeOutput.vue [hunk 1]
```diff
@@ -122,7 +122,7 @@ function onToggleActive() {
 
 function onToggleVisible() {
     const actionKey = `HideDatasetAction${props.output.name}`;
-    const step = stepStore.getStep(stepId.value);
+    const step = { ...stepStore.getStep(stepId.value) };
     assertDefined(step);
```
Comment (ElectronicBlueberry) [inline client/src/components/Workflow/Editor/NodeOutput.vue]: A more elegant solution than my previous suggestion.
Requires  #15834

```suggestion
    const step =  { ...ensureDefined(stepStore.getStep(stepId.value)) };
```

**Comment by martenson** [comment]
the rest of the test fails do not seem relevant to this merge -- I'm going forward with it

**Pull request merged**